indico/modules/core/client/js/impersonation.js

```diff
@@ -17,49 +17,56 @@
 
 /* global ChooseUsersPopup:false */
 
-(function() {
-    'use strict';
+import getImpersonateURL from 'indico-url:auth.admin_impersonate';
 
-    function sendRequest(url, payload) {
-        $.ajax({
-            url: url,
-            method: 'POST',
-            data: payload,
-            complete: IndicoUI.Dialogs.Util.progress(),
-            error: handleAjaxError,
-            success: function() {
-                IndicoUI.Dialogs.Util.progress();
-                location.reload();
-            }
-        });
-    }
+import {indicoAxios, handleAxiosError} from 'indico/utils/axios';
+import {$T} from 'indico/utils/i18n';
 
-    function impersonateUser(url) {
-        function _userSelected(users) {
-            sendRequest(url, {user_id: users[0].id});
-        }
 
-        var dialog = new ChooseUsersPopup(
-            $T("Select user to impersonate"),
-            true, null, false, true, null, true, true, false, _userSelected, null, false
-        );
+async function sendRequest(data) {
+    const formData = new FormData();
+    data.forEach(([k, v]) => {
+        formData.append(k, v);
+    });
 
-        dialog.execute();
+    try {
+        await indicoAxios.post(getImpersonateURL(), formData);
+    } catch (error) {
+        handleAxiosError(error);
+        return;
     }
+    window.location.reload();
+}
 
-    function undoImpersonateUser(url) {
-        sendRequest(url, {undo: '1'});
+function impersonateUser() {
+    function _userSelected(users) {
+        sendRequest([['user_id', users[0].id]]);
     }
 
-    $(document).ready(function() {
-        $('.login-as').on('click', function(evt) {
-            evt.preventDefault();
-            impersonateUser($(this).data('href'));
-        });
+    const dialog = new ChooseUsersPopup(
+        $T('Select user to impersonate'),
+        true, null, false, true, null, true, true, false, _userSelected, null, false
```

Comment:
:see_no_evil:  (i know, it was there before, but i almost forgot what an abomination the argument lists of those legacy widgets are)